### PR TITLE
fix: add windows specific ssh-config

### PR DESCRIPTION
**Pull request by Bexanderthebex**
This attempts to fix https://github.com/cloudflare/cloudflared/issues/753

The bug occurs because windows executable uses the current working directory to do its operations while the ssh-config being generated assumes that:

1. the user uses a nix machine
2. the nix executable follows a different installation directory than windows. Windows uses the relative path, the nix binary gets stored in `~/.cloudflared`

### CODE REVIEW

File: cmd/cloudflared/access/cmd.go [hunk 1]
```diff
@@ -39,13 +40,22 @@ Add to your {{.Home}}/.ssh/config:
...
+	{{- if eq .OS "windows"}}
+  IdentityFile {{.CurrWorkingDir}}/{{.Hostname}}-cf_key
+  CertificateFile {{.CurrWorkingDir}}/{{.Hostname}}-cf_key-cert.pub
+	{{- else}}
+	IdentityFile ~/.cloudflared/{{.Hostname}}-cf_key
   CertificateFile ~/.cloudflared/{{.Hostname}}-cf_key-cert.pub
+	{{end}}
 {{- else}}
   ProxyCommand {{.Cloudflared}} access ssh --hostname %h
 {{end}}
```
Comment (GreenStage) [inline cmd/cloudflared/access/cmd.go:61]:
```suggestion
{{- if eq .OS "windows"}}  IdentityFile {{.CurrWorkingDir}}/{{.Hostname}}-cf_key
  CertificateFile {{.CurrWorkingDir}}/{{.Hostname}}-cf_key-cert.pub
{{- else}}  IdentityFile ~/.cloudflared/{{.Hostname}}-cf_key
  CertificateFile ~/.cloudflared/{{.Hostname}}-cf_key-cert.pub
{{end}}
{{- else}}
  ProxyCommand {{.Cloudflared}} access ssh --hostname %h
{{end}}
```

Indentation is weird with go templates, we can't add whitespaces before template statements, and lets also replace tabs with double withspaces

File: cmd/cloudflared/access/cmd.go [hunk 2]
```diff
@@ -39,13 +40,22 @@ Add to your {{.Home}}/.ssh/config:
 
 Host {{.Hostname}}
 {{- if .ShortLivedCerts}}
+	{{- if eq .OS "windows"}}
+	ProxyCommand {{.Cloudflared}} access ssh-gen --hostname %h && ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1
+	{{- else}}
   ProxyCommand bash -c '{{.Cloudflared}} access ssh-gen --hostname %h; ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1'
+	{{end}}
```
Comment (GreenStage) [inline cmd/cloudflared/access/cmd.go:47]:
```suggestion
{{- if eq .OS "windows"}}
  ProxyCommand {{.Cloudflared}} access ssh-gen --hostname %h && ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1
{{- else}}
  ProxyCommand bash -c '{{.Cloudflared}} access ssh-gen --hostname %h; ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1'
{{end}}
```

**Comment by Bexanderthebex** [comment]
@GreenStage thank you for taking a look. I did the changes on my windows partition and forget about the spacing (I don't use it a lot) so thank you for noticing that.

File: cmd/cloudflared/access/cmd.go [hunk 3]
```diff
@@ -38,10 +39,20 @@ const (
 Add to your {{.Home}}/.ssh/config:
 
 {{- if .ShortLivedCerts}}
-Match host {{.Hostname}} exec "{{.Cloudflared}} access ssh-gen --hostname %h"
```
Comment (GreenStage) [inline cmd/cloudflared/access/cmd.go:41]: I think we still need this?

File: cmd/cloudflared/access/cmd.go [hunk 4]
```diff
@@ -38,10 +39,20 @@ const (
...
+{{- if eq .OS "windows"}}
+  ProxyCommand {{.Cloudflared}} access ssh-gen --hostname %h && ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1
```
Comment (GreenStage) [inline cmd/cloudflared/access/cmd.go:43]: I think you want `Match host {{.Hostname}} exec` instead of ProxyCommand here, no?

Comment (Bexanderthebex) [reply cmd/cloudflared/access/cmd.go:43]: how is that different than `ssh -tt %r@cfpipe-{{.Hostname}}`?

Comment (GreenStage) [reply cmd/cloudflared/access/cmd.go:43]: My bad, 
but you do need to specify that this proxy is only applying to the hostname in question.


```
Host cfpipe-{{.Hostname}}
  ProxyCommand {{.Cloudflared}} access ssh-gen --hostname %h && ssh -tt %r@cfpipe-{{.Hostname}} >&2 <&1
```

Comment (Bexanderthebex) [reply cmd/cloudflared/access/cmd.go:44]: Sorry for the late revert @GreenStage, just pushed the change that you mentioned. I wonder if that is good enough?

**Review by 01SMD — APPROVED** [review]

**Review by 01SMD — APPROVED** [review]

**Review by 01SMD — APPROVED** [review]

**Comment by Bexanderthebex** [comment]
@jroyal wonder if you can take a look?

**Comment by chungthuang** [comment]
Feel free to rebase and reopen it if it's still releveant.

**Pull request closed without merging**